tests/MongoDB.Driver.Core.Tests/Core/Clusters/DnsClientWrapperTests.cs

```diff
@@ -156,22 +147,34 @@ public void ResolveTxtRecords_should_throw_when_domainName_is_null(bool async)
         [InlineData(true)]
         public void ResolveTxtRecords_should_throw_when_cancellation_is_already_requested(bool async)
         {
-            var subject = new DnsClientWrapper();
+            var subject = CreateSubject();
             var domainName = "test5.test.build.10gen.cc";
             var cts = new CancellationTokenSource();
             cts.Cancel();
 
             Exception exception;
             if (async)
             {
-                exception = Record.Exception(() => subject.ResolveTxtRecords(domainName, cts.Token));
+                exception = Record.Exception(() => subject.ResolveTxtRecordsAsync(domainName, cts.Token).GetAwaiter().GetResult());
             }
             else
             {
-                exception = Record.Exception(() => subject.ResolveTxtRecordsAsync(domainName, cts.Token).GetAwaiter().GetResult());
+                exception = Record.Exception(() => subject.ResolveTxtRecords(domainName, cts.Token));
             }
 
             exception.Should().Match<Exception>(e => e is OperationCanceledException || e.InnerException is OperationCanceledException);
         }
+
+        // private methods
+        private IDnsResolver CreateSubject()
+        {
+            DnsClientWrapperReflector.__instance(null); // reset cache
+            return DnsClientWrapper.Instance;
+        }
+    }
+
+    internal static class DnsClientWrapperReflector
+    {
+        public static void __instance(IDnsResolver dnsResolver) => Reflector.SetStaticFieldValue(typeof(DnsClientWrapper), nameof(__instance), dnsResolver);
```

Comment:
Refactor to:

    internal static class DnsClientWrapperReflector
    {
        public static LookupClient _lookupClient(this DnsClientWrapper obj) => (LookupClient)Reflector.GetFieldValue(obj, nameof(_lookupClient));
    }

Reply:
done